op-service/sources/eth_client.go

```diff
@@ -547,12 +548,27 @@ func (s *EthClient) PendingNonceAt(ctx context.Context, account common.Address)
 	return uint64(result), err
 }
 
+func toBlockNumArg(number *big.Int) string {
```

Comment:
This is a very geth specific convention for what negative numbers mean.  I'm not sure how big a knock on effect it would have (probably enough it should be a separate PR), but I wonder if we should use [rpcblock.Block](https://github.com/ethereum-optimism/optimism/blob/70c0778dc1f2ee3addc733b264da3f0d3bf7761a/op-service/sources/batching/rpcblock/blocks.go#L10) which provides a nicer abstraction and keeps geth internal details out of the API.

Reply:
I like rpcblock.Block better as well, can separate out into a different PR.

Reply:
Actually, since this would involve modifying the EthClient interface (specifically the [EthBalance](https://github.com/ethereum-optimism/optimism/blob/507979b2606c075d30ef65f797545a2688a86dc4/op-service/apis/eth.go#L114) subinterface) which I suspect would require a larger change, would it be alright to make this smaller change for now to fix the existing code intent?

Reply:
Oh yeah sorry, I didn't mean that to be a blocking comment.  I don't really have the context to review this fully - just had the drive by suggestion of using `rpcblock`. :)